src/CNRECS.DEF

```diff
@@ -669,6 +669,56 @@ RECORD PY4 "4th power poly sub" *SUBSTRUCT	// 4th power polynomial. for GT, whic
     *f *ARRAY 6 FLOAT k			// FLOAT k[6].  One extra member needed as terminator by cul.cpp array input logic.
 *END		// PY4
 //==================================================================================================================
+RECORD PERFORMANCEMAP "PERFORMANCEMAP" *RAT
+
+    *prefix pm_
+    *declare "RC pm_CkF();"
+    *declare "RC pm_SetupBtwxt( record* pParent, const char* tag, class Btwxt::RegularGridInterpolator* &pRgi, double& capRatRef, double& ratingSpeedF) const;"
+    *declare "RC pm_GetGridAxis( int iGX, PMGRIDAXIS*& pGXRet) const;"
+    *declare "RC pm_GetLookupData( int iLU, PMLOOKUPDATA*& pLURet) const;"
+    *declare "RC pm_GetLookupValue( int iLU, float& luValRet, int iGx1, int iGx2) const;"
+    *declare "RC PERFORMANCEMAP::pm_GXCheckAndMakeVector( int iGX, PMGRIDAXIS* &pGX, std::vector< double>& vGX, std::pair< int, int> sizeLimits) const;"
+    *declare "RC PERFORMANCEMAP::pm_LUCheckAndMakeVector( int iLU, PMLOOKUPDATA* &pLU, std::vector< double>& vLU, int expectedSize, double vMin=-DBL_MIN, double vMax=DBL_MAX) const;"
+
+    *i SI pm_dummy      // *declare cannot be last (rcdef restriction / shortcoming)
+
+*END    // PERFORMANCEMAP
+//==================================================================================================================
+RECORD PMGRIDAXIS "PMGRIDAXIS" *RAT
+    *prefix pmx_
+
+    *ovrcopy
+    *declare "RC pmx_CkF();"
+    *declare "PERFORMANCEMAP* pmx_GetPERFMAP() const;"
+    *declare "RC pmx_CheckAndMakeVector( std::vector< double>& vGX, std::pair< int, int> sizeLimits) const;"
+
+    *i CULSTR pmx_id        // axis identifier (e.g. "DBT")
+                            // cannot use PMGRIDAXIS.name
+                            //   multiple PMGRIDAXISs may use same pmx_id
+    *i SI pmx_nValues       // # of pmx_values found in input
+    *i *ARRAY 7 FLOAT pmx_values      // axis values
+
+    *i FLOAT pmx_refValue   // nominal or reference value of axis
+                            //   e.g. 95 for cooling ODT
+                            //   used for normalizing and data consistency checking
+
+*END     // PMGRIDAXIS
+//=============================================================================
+RECORD PMLOOKUPDATA "PMLOOKUPDATA" *RAT
+    *prefix pmv_
+
+    *ovrcopy
+
+    *declare "RC pmv_CkF();"
+    *declare "PERFORMANCEMAP* pmv_GetPERFMAP() const;"
+    *declare "RC pmv_CheckAndMakeVector( std::vector< double>& vLU, int expectedsize, double vMin=-DBL_MAX, double vMax=DBL_MAX) const;"
+
+    *i CULSTR pmv_id        // data identifier
+    *i SI pmv_nValues       // # of pm_values found in input
+    *i *ARRAY 700 FLOAT_GEZ pmv_values   // lookup values in gridaxis order
```

Comment:
Make 1,000,000? Should be normal `FLOAT`.

Reply:
Due to fixed size arrays, the size limit is some guess at big enough.  Large sizes would use a lot of memory.  Issue added re using dynamically sized storage for *array.